cmd/pctl/search.go

```diff
@@ -25,42 +25,38 @@ func searchCmd() *cli.Command {
 				Value:       "table",
 				Usage:       "Output format. json|table",
 			},
+			&cli.BoolFlag{
+				Name:    "all",
+				Aliases: []string{"a"},
+				Usage:   "Search all available profiles",
+			},
 		},
 		Action: func(c *cli.Context) error {
-			searchName, catalogClient, err := parseArgs(c)
-			if err != nil {
-				_ = cli.ShowCommandHelp(c, "search")
-				return err
-			}
-
-			profiles, err := catalog.Search(catalogClient, searchName)
-			if err != nil {
-				return err
-			}
-			outFormat := c.String("output")
-			if outFormat == "table" {
-				if len(profiles) == 0 {
-					fmt.Printf("No profiles found matching: '%s'\n", searchName)
-					return nil
+			var profiles []profilesv1.ProfileCatalogEntry
+			if c.Bool("all") {
+				catalogClient, err := getClient(c)
+				if err != nil {
+					_ = cli.ShowCommandHelp(c, "search")
+					return err
+				}
+				profiles, err = catalog.Search(catalogClient, "all")
+				if err != nil {
+					return err
+				}
+			} else {
+				searchName, catalogClient, err := parseArgs(c)
```

Comment:
should we error if `--all` and an arg is provided?

Reply:
i am not sure , what do u think? atm i think if u specify `-all` and searchname it'll ignore the searchname and return all available profiles

Reply:
I think its worth erroring if both are set, to avoid confusion. People might think `--all` does something else otherwise

Reply:
i have added the error if both `-all` and searchname is provided